### PR TITLE
fix(gateway): propagate rejections to clients without retrying

### LINKED ISSUE #6166: Gateway keeps retrying rejections
**Describe the bug**
Related to this #5812. The gateway retries rejections (as well as any error) on different partitions and eventually returns a response with no jobs. This causes the clients to get "either timeouts or resource exhausted responses" which masks the true error and results in poor user experience and performance.

**Expected behavior**
The treats rejections as non-transient errors and returns them to the clients.

**Environment:**
 - OS: [e.g. Linux]
 - Zeebe Version: [e.g. 0.20.0]
 - Configuration: [e.g. exporters etc.]

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Previously, a rejected job activate request would be ignore and retried in the remaining partitions. In the end, the rejection would be suppressed and a response with zero jobs would be returned. Since rejections are most likely not transient errors, there's not point in retrying. Furthermore, returning zero jobs will cause the clients to eventually retry. This PR changes the gateway to not retry on rejections and actually return the rejection as an error. This will allow the clients to detect this and react accordingly (issues camunda/camunda#6167 camunda/issue-migration-go-client#14).

## Related issues

closes camunda/camunda#6166 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: gateway/src/main/java/io/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java [hunk 1]
```diff
@@ -190,12 +195,11 @@ private void resetFailedAttemptsAndHandlePendingRequests(final String jobType) {
     final Queue<LongPollingActivateJobsRequest> pendingRequests = state.getPendingRequests();
 
     if (!pendingRequests.isEmpty()) {
-      pendingRequests.stream()
-          .forEach(
-              nextPendingRequest -> {
-                LOG.trace("Unblocking ActivateJobsRequest {}", nextPendingRequest.getRequest());
-                activateJobs(nextPendingRequest);
-              });
+      pendingRequests.forEach(
```
Comment (pihme) [inline gateway/src/main/java/io/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java:198]: nice catch

File: gateway/src/main/java/io/zeebe/gateway/impl/job/RoundRobinActivateJobsHandler.java [hunk 2]
```diff
@@ -66,13 +68,15 @@ public void activateJobs(
       final int maxJobsToActivate,
       final String type,
       final Consumer<ActivateJobsResponse> onResponse,
+      final Consumer<Throwable> onError,
```
Comment (pihme) [inline gateway/src/main/java/io/zeebe/gateway/impl/job/RoundRobinActivateJobsHandler.java:71]: Optional: Maybe rename to `onErrorConsumer`/`onErrorHandler` (dito for` onNext/Completed`)

File: gateway/src/test/java/io/zeebe/gateway/api/job/ActivateJobsTest.java [hunk 3]
```diff
@@ -138,4 +149,33 @@ public void shouldActivateJobsRoundRobin() {
...
+
+    // when/then
+    assertThatThrownBy(
+            () -> {
+              final Iterator<ActivateJobsResponse> responseIterator = client.activateJobs(request);
+              responseIterator.hasNext();
+            })
+        .isInstanceOf(StatusRuntimeException.class)
+        .extracting(t -> ((StatusRuntimeException) t).getStatus().getCode())
+        .matches(code -> code == Status.INVALID_ARGUMENT.getCode());
```
Comment (pihme) [inline gateway/src/test/java/io/zeebe/gateway/api/job/ActivateJobsTest.java:178]:
```suggestion
        .isEqualTo(Status.INVALID_ARGUMENT.getCode());
```

File: gateway/src/test/java/io/zeebe/gateway/api/job/ActivateJobsTest.java [hunk 4]
```diff
@@ -138,4 +149,33 @@ public void shouldActivateJobsRoundRobin() {
       }
     }
   }
+
+  @Test
+  public void shouldRejectWithoutRetrying() {
```
Comment (pihme) [inline gateway/src/test/java/io/zeebe/gateway/api/job/ActivateJobsTest.java:154]:
```suggestion
  public void shouldSendRejectionWithoutRetrying() {
```

**Review by pihme — APPROVED** [review]
Nice work!

Some suggestions (see comments). Feel free to adopt or ignore them.

From my side, I don't need to look at the review again if you implement some of the suggestion. But happy to look at it again, if you want me to.

One final request, though: Please add the paragraph you have for this PR to the release notes in the bugfixes section. Maybe rephrase it a bit. 

I think the release notes for the next release are still called 0.27.0 in the folder.

**Comment by MiguelPires** [comment]
Thanks for the review. I squashed the suggestions into the commit. I'll add this and some other bug fixes to the release notes, thanks for reminding me
